src/ripple/app/tx/impl/ApplyContext.cpp

```diff
@@ -103,17 +103,16 @@ ApplyContext::checkInvariantsHelper(
                     std::shared_ptr <SLE const> const& before,
                     std::shared_ptr <SLE const> const& after)
                 {
-                    // Sean Parent for_each_argument trick
-                    (void)std::array<int, sizeof...(Is)>{
-                        {((std::get<Is>(checkers).
-                            visitEntry(index, isDelete, before, after)), 0)...}
-                    };
+                    (..., std::get<Is>(checkers).visitEntry(index, isDelete, before, after));
                 });
 
-            // Sean Parent for_each_argument trick (a fold expression with `&&`
-            // would be really nice here when we move to C++-17)
-            std::array<bool, sizeof...(Is)> finalizers {{
-                std::get<Is>(checkers).finalize(tx, result, fee, journal)...}};
+            // Note: do not replace this logic with a `...&&` fold expression.
```

Comment:
Nice comment!  Thanks!

Reply:
Nitpick on the comment:  "The fold expression" is ambiguous.  It could refer to `...,` or `...&&`.  How about:

// The `...&&` fold expression will ...